inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp

```diff
@@ -411,26 +423,96 @@ static void TransformationUpToCPUSpecificOpSet(std::shared_ptr<ngraph::Function>
         });
         lptManager.run_passes(nGraphFunc);
     }
+    ngraph::pass::Manager postCommonPassManager;
+    postCommonPassManager.register_pass<ngraph::pass::FakeQuantizeDecomposition>();
+    postCommonPassManager.register_pass<ngraph::pass::UnrollTensorIterator>();
+    postCommonPassManager.register_pass<ReshapePRelu>();
 
-    ngraph::pass::Manager postLPTPassManager;
-    postLPTPassManager.register_pass<ngraph::pass::FakeQuantizeDecomposition>();
-    postLPTPassManager.register_pass<ngraph::pass::UnrollTensorIterator>();
-
-    postLPTPassManager.get_pass_config()->set_callback<ngraph::pass::FakeQuantizeDecomposition>([](const_node_ptr &node) -> bool {
+    postCommonPassManager.get_pass_config()->set_callback<ngraph::pass::FakeQuantizeDecomposition>([](const_node_ptr &node) -> bool {
         std::string errMsg;
         return MKLDNNFakeQuantizeNode::isSupportedOperation(node, errMsg);
     });
-    postLPTPassManager.get_pass_config()->set_callback<ngraph::pass::UnrollTensorIterator>([](const_node_ptr &node) -> bool {
+    postCommonPassManager.get_pass_config()->set_callback<ngraph::pass::UnrollTensorIterator>([](const_node_ptr &node) -> bool {
         // UnrollTI transformation is disabled by default, is turned on by LowLatency transformation
         return node->get_rt_info().count("UNROLL_TI") == 0;
     });
 
-    postLPTPassManager.run_passes(nGraphFunc);
+    postCommonPassManager.run_passes(nGraphFunc);
+
+    // todo: implement a more precise check for BF16? Traverse graph and check precisions?
+    const bool enableBF16  = _enforceBF16 && with_cpu_x86_avx512_core();
```

Comment:
It certainly would, corrected.

Reply:
Update: a brief research showed that [smoke_FP32_bfloat16_NoReshape](https://github.com/openvinotoolkit/openvino/blob/fa2f9c5201983d7ce247d1723083ca033f3c672d/inference-engine/tests/functional/plugin/cpu/bfloat16/gather_x2_add_mul_relu_concat_matmul.cpp#L140-L147) fails if we enableBF16 only on with_cpu_x86_bfloat16 (but not on  with_cpu_x86_avx512_core). This happens because bf16 tests are executed  on all avx512_core in the ["simulation mode"](https://github.com/IvanNovoselov/openvino/blob/fa2f9c5201983d7ce247d1723083ca033f3c672d/inference-engine/tests/functional/plugin/cpu/bfloat16/bfloat16_helpers.hpp#L182-L186). So i think I'm gonna stick with the original with_cpu_x86_avx512_core() as well (or we need to disable the bf16 "simulation mode").